### PR TITLE
Will/lockfile pnpm8 support

### LINKED ISSUE #4254: [lockfile-explorer] Lockfile v6 support
Self-explanatory. Doesn't seem to be there right now.

### PR DESCRIPTION
**Pull request by william2958**
## Summary
This MR adds the ability for lockfile explorer to read pnpm 8 lockfiles correctly (pnpm lockfiles that are using the v6 spec)

Fixes #4254

## Details
v6 of the pnpm lockfile slightly changes the way entries are recorded, replacing the splitter between the package and it's version with a '@' instead of a '/'. This MR just adapts the new format to the old format so the explorer can read the lockfile correctly.

## How it was tested
Tested manually

### CODE REVIEW

File: apps/lockfile-explorer-web/src/parsing/readLockfile.ts [hunk 1]
```diff
@@ -16,22 +21,36 @@ export interface IPackageJsonType {
     [key in string]: string;
   };
 }
-
-export interface ILockfilePackageType {
-  lockfileVersion: number;
-  importers?: {
+export interface ILockfileImporterV6Type {
```
Comment (iclanton) [inline apps/lockfile-explorer-web/src/parsing/readLockfile.ts:24]:
```suggestion
export interface ILockfileImporterV6 {
```

File: apps/lockfile-explorer-web/src/parsing/readLockfile.ts [hunk 2]
```diff
@@ -16,22 +21,36 @@ export interface IPackageJsonType {
...
+    };
+  };
+  devDependencies?: {
+    [key in string]: {
+      specifier: string;
+      version: string;
     };
   };
+}
+export interface ILockfileImporterV5Type {
```
Comment (iclanton) [inline apps/lockfile-explorer-web/src/parsing/readLockfile.ts:38]:
```suggestion
export interface ILockfileImporterV5 {
```

File: apps/lockfile-explorer-web/src/parsing/readLockfile.ts [hunk 3]
```diff
@@ -16,22 +21,36 @@ export interface IPackageJsonType {
...
+    [key in string]: string;
+  };
+  devDependencies?: {
+    [key in string]: string;
+  };
+}
+export interface ILockfilePackageType {
+  lockfileVersion: number | string;
+  importers?: {
+    [key in string]: ILockfileImporterV5Type | ILockfileImporterV6Type;
```
Comment (iclanton) [inline apps/lockfile-explorer-web/src/parsing/readLockfile.ts:52]:
```suggestion
    [key in string]: ILockfileImporterV5 | ILockfileImporterV6;
```

File: common/changes/@rushstack/lockfile-explorer/will-lockfile-pnpm8-support_2023-08-31-23-03.json [hunk 4]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@rushstack/lockfile-explorer",
+      "comment": "Add support for pnpm8's lockfile v6",
```
Comment (iclanton) [inline common/changes/@rushstack/lockfile-explorer/will-lockfile-pnpm8-support_2023-08-31-23-03.json:5]:
```suggestion
      "comment": "Add support for V6 lockfile used by PNPM 8.",
```

File: apps/lockfile-explorer-web/src/parsing/readLockfile.ts (hunk 1 above)
Comment (william2958) [reply apps/lockfile-explorer-web/src/parsing/readLockfile.ts]: Renamed

File: apps/lockfile-explorer-web/src/parsing/readLockfile.ts (hunk 2 above)
Comment (william2958) [reply apps/lockfile-explorer-web/src/parsing/readLockfile.ts]: Renamed

File: apps/lockfile-explorer-web/src/parsing/readLockfile.ts (hunk 3 above)
Comment (william2958) [reply apps/lockfile-explorer-web/src/parsing/readLockfile.ts]: Renamed

**Review by iclanton — APPROVED** [review]

**Pull request merged**